pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -267,39 +269,82 @@ func (s *NodeController) DoCheck(node *api.Node) []api.NodeCondition {
 	// Check Condition: NodeReady. TODO: More node conditions.
 	oldReadyCondition := s.getCondition(node, api.NodeReady)
 	newReadyCondition := s.checkNodeReady(node)
-	if oldReadyCondition != nil && oldReadyCondition.Status == newReadyCondition.Status {
-		newReadyCondition.LastTransitionTime = oldReadyCondition.LastTransitionTime
-	} else {
+	if oldReadyCondition == nil {
+		// 'oldReadyCondition' can be nil only if the node is checked for the first time.
 		newReadyCondition.LastTransitionTime = util.Now()
+	} else {
+		if oldReadyCondition.Status == newReadyCondition.Status {
+			// If node status doesn't change, transition time is same as last time.
+			newReadyCondition.LastTransitionTime = oldReadyCondition.LastTransitionTime
+		} else {
+			// If node status changes this time, tansition time is last prob time, i.e.
+			// if it changes from ready -> not ready, we consider the interval between
+			// the transition as not ready.
+			// FIXME: if we consider the interval to be ready, then we can eliminate the new field.
+			newReadyCondition.LastTransitionTime = oldReadyCondition.LastProbTime
```

Comment:
ok, if we include LastProbeTime, then we'll keep updating status to apiserver.  I removed the code  where we skip the update.

I'll open a PR to include the LastProbeTime in client.